networkx/algorithms/distance_measures.py

```diff
@@ -815,7 +816,12 @@ def barycenter(G, weight=None, attr=None, sp=None):
     --------
     center
     periphery
+    :func:`~networkx.algorithms.tree.distance_measures.tree_centroid`
     """
+    if weight is None and attr is None and sp is None:
+        if nx.is_tree(G) and not G.is_directed():
+            return nx.tree_centroid(G)
```

Comment:
In case it doesn't get exposed in the main namespace.
```suggestion
            return nx.tree.tree_centroid(G)
```